client/ayon_core/tools/loader/ui/product_types_widget.py

```diff
@@ -162,7 +163,26 @@ def _on_project_change(self, event):
         project_name = event["project_name"]
         self._product_types_model.refresh(project_name)
 
+    def showEvent(self, event):
+        self._refresh_product_types_filter = True
+        super().showEvent(event)
+
     def _on_refresh_finished(self):
+
+        # Apply product types filter
+        if self._refresh_product_types_filter:
```

Comment:
BTW I would say this should happen in the model instead.

Reply:
How would that occur - if it's the viewer that needs to update the selection model?

Reply:
Hmm - it can just be moved to the `product_types_model` since we're using functions of that anyway.
Should I hence make a `update_product_type_filters_for_project` to the model so the view can still trigger it on the show event?

Reply:
Well, what is the point of using view? To refresh the settings on showing it? If yes then just add method `reset_product_type_filter` to the model to refresh it on next `refresh`?

Reply:
Done with https://github.com/ynput/ayon-core/pull/792/commits/88959b2c54a5a9569d9e426a29587bd829c3b72f